fedimint-cli/src/lib.rs

```diff
@@ -1308,6 +1322,38 @@ impl FedimintCli {
     }
 }
 
+async fn log_expiration_notice(client: &Client) {
+    client.get_meta_expiration_timestamp().await;
+    if let Some(expiration_time) = client.get_meta_expiration_timestamp().await {
+        match expiration_time.duration_since(fedimint_core::time::now()) {
+            Ok(until_expiration) => {
+                let days = until_expiration.as_secs() / (60 * 60 * 24);
+
+                if 90 < days {
+                    debug!(target: LOG_CLIENT, %days, "This federation will expire");
+                } else if 30 < days {
+                    info!(target: LOG_CLIENT, %days, "This federation will expire");
+                } else {
+                    warn!(target: LOG_CLIENT, %days, "This federation will expire soon");
+                }
+            }
+            Err(_) => {
+                tracing::error!(target: LOG_CLIENT, "This federation has expired and might not be safe to use");
+            }
+        }
+    }
+}
+async fn print_welcome_message(client: &Client) {
+    if let Some(welcome_message) = client
+        .meta_service()
+        .get_field::<String>(client.db(), "welcome_message")
+        .await
+        .and_then(|v| v.value)
+    {
+        eprintln!("{welcome_message}");
```

Comment:
Should this maybe be an info log too? Feels weird having a random `eprintln!(…)` for this one case.

Reply:
It makes it look ugly though... . Since it's only inside `fedimint-cli` the binary, the `fedimint-client`, I think it's acceptable to make an exception for it. @elsirion 

We can always change our minds later. I just thought bringing up some of the meta handling to `fedimnit-cli` will be a good example, test for devs and educational tool for the user.